### ISSUE TITLE
Gluey 2: Cannot start game

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#11210]
### Describe the bug

This issue applies to both versions of the game, which can be found at these URLs:
https://www.coolmath-games.com/sites/cmatgame/files/games/gluey2-Events.swf
http://i.notdoppler.com/files/gluey2.swf

When trying to start the game, the Coolmath version shows a black screen, while the NotDoppler version's intro loops indefinitely.

The NotDoppler intro loops due to a mixed-avm issue; the intro is embedded as a separate SWF loaded via BinaryData, and that SWF uses AVM1, with a stop() action on the last frame.

But both versions of the game are unable to start, showing the following error:
```
ERROR render: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "render", class: flash.events::Event, ptr: 0x253e2878a68 }) to handler FunctionObject(FunctionObject { ptr: 0x253e1b05268 }) : TypeError: Error #1006: value is not a function.
        at mx.core::Container/createComponentFromDescriptor()
        at mx.core::Container/createComponentsFromDescriptors()
        at mx.core::Container/createChildren()
        at mx.core::UIComponent/initialize()
        at mx.core::Container/initialize()
        at mx.core::UIComponent/http://www.adobe.com/2006/flex/mx/internal::childAdded()
        at mx.core::Container/http://www.adobe.com/2006/flex/mx/internal::childAdded()
        at mx.core::Container/addChildAt()
        at mx.core::Container/addChild()
        at mx.core::Container/createComponentFromDescriptor()
        at mx.core::Container/createComponentsFromDescriptors()
        at mx.core::Container/createChildren()
        at mx.core::UIComponent/initialize()
        at mx.core::Container/initialize()
        at com.fizzgear.game.mxmls.screens::SitelockMainMenu/initialize()
        at mx.core::UIComponent/http://www.adobe.com/2006/flex/mx/internal::childAdded()
        at mx.core::Container/http://www.adobe.com/2006/flex/mx/internal::childAdded()
        at mx.core::Container/addChildAt()
        at mx.core::Container/addChild()
        at _-6o::_-6L/commitProperties()
        at mx.core::UIComponent/validateProperties()
        at mx.managers::LayoutManager/validateProperties()
        at mx.managers::LayoutManager/doPhasedInstantiation()
        at Function/apply()
        at mx.core::UIComponent/callLaterDispatcher2()
        at mx.core::UIComponent/callLaterDispatcher()
```
This is the more important problem because it prevents both versions of the game from working.

### Expected behavior

The game should start. And the NotDoppler intro should not loop.

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

`Error #1006: value is not a function` is also present in #1623, another game created with Flex.

**Comment by Lord-McSweeney** [comment]
I believe the "value is not a function" (also present on http://cache.armorgames.com/files/games/armor-mayhem-chronic-14230.swf?v=1434137664) might be caused by a failing primitive coercion which works correctly in FP.

**Comment by n0samu** [comment]
The NotDoppler version of the game is working fine now, besides the annoying looping intro sound (mixed-avm issue). Meanwhile the Coolmath version is still showing a black screen; the error is:
`ERROR render: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "render", class: flash.events::Event, ptr: 0x28792b691b0 }) to handler FunctionObject(FunctionObject { ptr: 0x2878de82090 }) : RustError("Failed to lookup class Multiname { ns: Single(Namespace(Namespace(\"\"))), name: Some(\"void\"), param: None, flags: MultinameFlags(0x0) } during coercion")`

**Comment by n0samu** [comment]
Thanks to #11520 both versions of the game are playable, so I'm closing the issue.

**Comment by n0samu** [comment]
Looping intro sound is fixed thanks to #11005!
